python/cudf/cudf/tests/test_reshape.py

```diff
@@ -117,6 +117,98 @@ def test_df_stack(nulls, num_cols, num_rows, dtype):
     pass
 
 
+@pytest.mark.debug
+@pytest.mark.parametrize("num_rows", [1, 2, 10, 1000])
+@pytest.mark.parametrize("num_cols", [1, 2, 10])
+@pytest.mark.parametrize(
+    "dtype",
+    [
+        "int8",
+        "int16",
+        "int32",
+        "int64",
+        "float32",
+        "float64",
+        "datetime64[ms]",
+        "category",
+    ],
+)
+@pytest.mark.parametrize("nulls", ["none", "some"])
+def test_interleave_columns(nulls, num_cols, num_rows, dtype):
+
+    from pandas.api.types import CategoricalDtype
+
+    if dtype not in ["float32", "float64"] and nulls in ["some"]:
+        pytest.skip(msg="nulls not supported in dtype: " + dtype)
+
+    if dtype == "category":
+        dtype = CategoricalDtype(categories=range(num_cols), ordered=True)
+
+    pdf = pd.DataFrame(dtype=dtype)
```

Comment:
You can't pass the cudf `CategoricalDtype` to a pandas function.

Reply:
fixed